apps/channels/audio.py

```diff
@@ -17,6 +20,60 @@ def get_transcript(audio: BytesIO) -> str:
 
 
 def synthesize_voice(text: str, synthetic_voice: SyntheticVoice) -> Tuple[BytesIO, float]:
+    if synthetic_voice.service == "AWS":
+        return AWS_synthesize_voice(text, synthetic_voice)
+    elif synthetic_voice.service == "azure":
+        return azure_synthesize_voice(text, synthetic_voice)
+
+    raise AudioSynthesizeException(f"Unrecognized audio service: {synthetic_voice.service}")
+
+
+def azure_synthesize_voice(text: str, synthetic_voice: SyntheticVoice) -> Tuple[BytesIO, float]:
+    """
+    Calls Azure's cognitive speech services to convert the text to speech using the synthetic_voice
+    """
+    try:
+        speech_config = speechsdk.SpeechConfig(
+            subscription=settings.AZURE_SUBSCRIPTION_KEY, region=settings.AZURE_REGION
+        )
+
+        temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=".wav")
+
+        # Create an audio configuration that points to the output audio file
+        audio_config = speechsdk.audio.AudioConfig(filename=temp_file.name)
+
+        # Setup voice font (Azure refers to synthetic voices as voice fonts)
+        speech_config.speech_synthesis_voice_name = f"{synthetic_voice.language_code}-{synthetic_voice.name}"
+
+        # Create a speech synthesizer
+        synthesizer = speechsdk.SpeechSynthesizer(speech_config=speech_config, audio_config=audio_config)
+
+        # Synthesize the text
+        result = synthesizer.speak_text_async(text).get()
+
+        # Check if synthesis was successful
+        if result.reason == speechsdk.ResultReason.SynthesizingAudioCompleted:
+            audio_segment = AudioSegment.from_file(temp_file.name, format="wav")  # Azure returns audio in WAV format
+            duration_seconds = len(audio_segment) / 1000  # Convert milliseconds to seconds
+
+            with open(temp_file.name, "rb") as f:
+                file_content = f.read()
+
+            # Delete the temporary file
+            os.unlink(temp_file.name)
+
+            return BytesIO(file_content), duration_seconds
+        else:
+            raise AudioSynthesizeException(f"Azure audio synthesis failed with reason: {result.reason}. Req")
+
+    except Exception as e:
+        raise AudioSynthesizeException(f"Unable to synthesize audio with Azure: {e}")
+
+
+def AWS_synthesize_voice(text: str, synthetic_voice: SyntheticVoice) -> Tuple[BytesIO, float]:
```

Comment:
typically function names are in snake case

Reply:
I wonder if we should add that check to the pre-commit formatting stuff? Anyway, https://github.com/dimagi/open-chat-studio/pull/48/commits/3758fbb3f1294522021fad16d44f0096057d867a

Reply:
Not sure if the lemon would be worth the squeeze? But do feel free to do it if you're up for it. I think a review will quickly sniff em' stray names out :smile: